shell/platform/android/test/io/flutter/plugin/editing/SpellCheckPluginTest.java

```diff
@@ -0,0 +1,132 @@
+    TextServicesManager fakeTextServicesManager = mock(TextServicesManager.class);
+    when(fakeContext.getSystemService(Context.TEXT_SERVICES_MANAGER_SERVICE))
+        .thenReturn(fakeTextServicesManager);
+    SpellCheckPlugin spellCheckPlugin = new SpellCheckPlugin(fakeContext, fakeSpellCheckChannel);
+    SpellCheckerSession fakeSpellCheckerSession = mock(SpellCheckerSession.class);
+    when(fakeTextServicesManager.newSpellCheckerSession(
+            null, new Locale("en", "US"), spellCheckPlugin, true))
+        .thenReturn(fakeSpellCheckerSession);
+
+    spellCheckPlugin.performSpellCheck("en-US", "Hello, wrold!");
+    spellCheckPlugin.destroy();
+
+    verify(fakeSpellCheckChannel).setSpellCheckMethodHandler(isNull());
+    verify(fakeSpellCheckerSession).close();
+  }
+
+  @Test
+  public void performSpellCheckSendsRequestToAndroidSpellCheckService() {
+    Context fakeContext = mock(Context.class);
+    SpellCheckChannel fakeSpellCheckChannel = mock(SpellCheckChannel.class);
+    TextServicesManager fakeTextServicesManager = mock(TextServicesManager.class);
+    when(fakeContext.getSystemService(Context.TEXT_SERVICES_MANAGER_SERVICE))
+        .thenReturn(fakeTextServicesManager);
+    SpellCheckPlugin spellCheckPlugin = new SpellCheckPlugin(fakeContext, fakeSpellCheckChannel);
+    SpellCheckerSession fakeSpellCheckerSession = mock(SpellCheckerSession.class);
+    when(fakeTextServicesManager.newSpellCheckerSession(
+            null, new Locale("en", "US"), spellCheckPlugin, true))
+        .thenReturn(fakeSpellCheckerSession);
+
+    ArgumentCaptor<TextInfo[]> textInfosCaptor = ArgumentCaptor.forClass(TextInfo[].class);
+    ArgumentCaptor<Integer> maxSuggestionsCaptor = ArgumentCaptor.forClass(Integer.class);
+
+    spellCheckPlugin.performSpellCheck("en-US", "Hello, wrold!");
+
+    verify(fakeSpellCheckerSession)
+        .getSentenceSuggestions(textInfosCaptor.capture(), maxSuggestionsCaptor.capture());
+    assertEquals("Hello, wrold!", textInfosCaptor.getValue()[0].getText());
+    assertEquals(Integer.valueOf(3), maxSuggestionsCaptor.getValue());
+  }
+
+  @Test
+  public void onGetSentenceSuggestionsProperlyRequestsUpdateSpellCheckResults() {
+    Context fakeContext = mock(Context.class);
+    SpellCheckChannel fakeSpellCheckChannel = mock(SpellCheckChannel.class);
+    SpellCheckPlugin spellCheckPlugin = new SpellCheckPlugin(fakeContext, fakeSpellCheckChannel);
+
+    spellCheckPlugin.onGetSentenceSuggestions(
+        new SentenceSuggestionsInfo[] {
+          new SentenceSuggestionsInfo(
+              (new SuggestionsInfo[] {
+                new SuggestionsInfo(
+                    SuggestionsInfo.RESULT_ATTR_LOOKS_LIKE_TYPO,
+                    new String[] {"world", "word", "old"})
+              }),
+              new int[] {7},
+              new int[] {5})
+        });
+
+    verify(fakeSpellCheckChannel)
+        .updateSpellCheckResults(new ArrayList<String>(Arrays.asList("7.11.world,word,old")));
+  }
+}
```

Comment:
Nit: Would it be useful to also test getting spell check suggestions multiple times?  Like send "hello worl" and then test "hello world" and verify that the session gets reset?